include/mbedtls/ssl.h

```diff
@@ -798,7 +814,14 @@ struct mbedtls_ssl_session
     unsigned char master[48];   /*!< the master secret  */
 
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
-    mbedtls_x509_crt *peer_cert;        /*!< peer X.509 cert chain */
+    mbedtls_x509_crt *peer_cert;       /*!< peer X.509 cert chain */
+#if !defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
+    /*! The digest of the peer's end-CRT. This must be kept to detect CRT
+     *  changes during renegotiation, mitigating the triple handshake attack. */
+    unsigned char *peer_cert_digest;
```

Comment:
@mpg I agree with you, it would be better to embed the digest here; that would also make session serialization simpler. That said, however, that might lead to larger changes in the session ticket module - I'll have a look.

Reply:
Yes, actually I think it's probably safer not to make such changes at this point, but I just wanted to point that out for the future.

Reply:
Modifying `mbedtls_ssl_session` is an API break - can we change the API in a non-default configuration? If so, I'd probably prefer to leave the change for later; if not, let's do it now.

Reply:
I think it's a grey area. I'd rather leave the change for later (we already have an item to optimize structures in our internal pending API changes list, feel free to expand it with info about this) than having a discussion about what's an acceptable API change under time pressure.